internal/watcher/health/health_watcher_service.go

```diff
@@ -98,12 +109,19 @@ func (hw *HealthWatcherService) Watch(ctx context.Context, ch chan<- InstanceHea
 	}
 }
 
-func (hw *HealthWatcherService) health(ctx context.Context) (healthStatuses []*v1.InstanceHealth, isHealthDiff bool,
+func (hw *HealthWatcherService) health(ctx context.Context) (healthStatuses []*mpi.InstanceHealth, isHealthDiff bool,
 ) {
-	currentHealth := make(map[string]*v1.InstanceHealth, len(hw.watchers))
+	currentHealth := make(map[string]*mpi.InstanceHealth, len(hw.watchers))
 
 	for instanceID, watcher := range hw.watchers {
-		instanceHealth := watcher.Health(ctx, instanceID)
+		instanceHealth, err := watcher.Health(ctx, hw.instances[instanceID])
+		if instanceHealth == nil {
+			instanceHealth = &mpi.InstanceHealth{
+				InstanceId:           instanceID,
+				InstanceHealthStatus: mpi.InstanceHealth_INSTANCE_HEALTH_STATUS_UNSPECIFIED,
+				Description:          fmt.Sprintf("failed to get health for instance %s, error: %v", instanceID, err),
+			}
+		}
 		healthStatuses = append(healthStatuses, instanceHealth)
```

Comment:
Could we perhaps not use the `healthStatuses` slice here and init like a new slice for readability purposes.

So this could be something like a `allStatuses` slice and then on line 135 we could have a `updatedStatuses` slice? Just cuz imo it's harder to follow with only the one slice in the current implementation.

Reply:
I think I have made those changes if you could have another look. Not 100% sure I understood the comment correctly. Thank you